api/envoy/type/matcher/v3/http_inputs.proto

```diff
@@ -60,3 +60,15 @@ message HttpResponseTrailerMatchInput {
   string header_name = 1
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false}];
 }
+
+// Match input indicates that matching should be done on a specific query parameter.
+// The resulting input string will be the first query parameter for the value
+// 'query_param'.
+// [#extension: envoy.matching.inputs.query_params]
+message HttpRequestQueryParamMatchInput {
+  // The query parameter to match on.
+  string query_param = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Whether the query parameters and their value should be URL-decoded.
+  bool url_decoded = 2;
```

Comment:
In production, it seems is always safe and legal to use `parseAndDecodeQueryString` for URL (but not for `parseQueryString` doesn't handle percent-encoding)

So, do we need `bool url_decoded` here in the proto? Can we remove this flag and just always use `parseAndDecodeQueryString`
WDYT?

Reply:
I think @tzuryby has more inputs here: what is used from a WAF point of view, decoded or raw query strings?

Reply:
Parse and decode. 
This is the value the waf will match.

Reply:
Fixed here https://github.com/envoyproxy/envoy/pull/24627/commits/5305a220ce918ae6fbac237d031bcb97fed2ff13#diff-8ab5a462143e5ab5a9fa350b07d8a8c86b98a63a648066cba38cca1d1fe76638L71